src/lib/autocomplete/autocomplete-trigger.ts

```diff
@@ -201,7 +201,15 @@ export class MatAutocompleteTrigger implements ControlValueAccessor, OnDestroy {
 
   /** Stream of autocomplete option selections. */
   get optionSelections(): Observable<MatOptionSelectionChange> {
-    return merge(...this.autocomplete.options.map(option => option.onSelectionChange));
+    if (this.autocomplete && this.autocomplete.options) {
```

Comment:
That would add some extra bloat since we'd have to maintain another property, as well drop all subscriptions and re-subscribe as the amount of options changes. Since this is mostly a convenience API, I'm not sure that it's worth the trouble.

Reply:
You probably wanted to do:

```ts
optionSelections = defer(() => {
  if (this.autoComplete && this.autoComplete.options) {
   return merge(...this.autocomplete.options.map(option => option.onSelectionChange));
  }
  return this._zone.onStable
      .asObservable()
      .pipe(take(1), switchMap(() => this.optionSelections));
});
```

Remember, Observables are basically functions, you wouldn't create a getter that returned a new function reference every time. `defer` is just a nice trick for making sure whatever the observable needs to be created exists when you subscribe to the observable.